charts/nym-mixnode/templates/deployment.yaml

```diff
@@ -0,0 +1,92 @@
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: {{ include "mixnode.fullname" . }}
+  labels:
+    {{- include "mixnode.labels" . | nindent 4 }}
+spec:
+  replicas: {{ .Values.replicaCount }}
+  selector:
+    matchLabels:
+      {{- include "mixnode.selectorLabels" . | nindent 6 }}
+  template:
+    metadata:
+      labels:
+        {{- include "mixnode.selectorLabels" . | nindent 8 }}
+    spec:
+    {{- with .Values.imagePullSecrets }}
+      imagePullSecrets:
+        {{- toYaml . | nindent 8 }}
+    {{- end }}
+      serviceAccountName: {{ include "mixnode.serviceAccountName" . }}
+      securityContext:
+        {{- toYaml .Values.podSecurityContext | nindent 8 }}
+{{ if eq .Values.environment "production" }}
+      initContainers:
+        - name: init-config
+          image: "{{ .Values.image.repository }}:{{ .Values.image.tag }}"
+          imagePullPolicy: {{ .Values.image.pullPolicy }}
+          env:
+          - name: INTERNAL_IP
+            valueFrom:
+              fieldRef:
+                fieldPath: status.podIP
+          volumeMounts:
+          - name: config-volume
+            mountPath: /root/.nym/mixnodes/w3f/config/
+          - name: gcloud-volume
+            mountPath: /root/.gcloud
+            readOnly: true
+          command: ["/bin/sh", "-c"]
+          args:
+            - gcloud auth activate-service-account --key-file=/root/.gcloud/credentials.json;
```

Comment:
No please, we won't give this pod running an external application full power over our infrastructure; we need to remove gcloud, create a role that only is allowed to read services properties and a role binding that associates that role with the pod service account. Then kubectl alone is going to be able to get just the required information.